unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/AssetManager/AssetBundles/SceneAB/AssetPromise_SceneAB.cs

```diff
@@ -18,14 +18,28 @@ public class SceneAbDto
 
     public class AssetPromise_SceneAB : AssetPromise_WithUrl<Asset_SceneAB>
     {
+        private const string URN_PREFIX = "urn:decentraland:entity:";
         private readonly CancellationTokenSource cancellationTokenSource;
         private Service<IWebRequestController> webRequestController;
 
         private Action onSuccess;
 
-        public AssetPromise_SceneAB(string contentUrl, string hash) : base(contentUrl, hash)
+        public AssetPromise_SceneAB(string contentUrl, string sceneId) : base(contentUrl, sceneId)
         {
             cancellationTokenSource = new CancellationTokenSource();
+            this.hash = GetEntityIdFromSceneId(sceneId);
+        }
+
+        private string GetEntityIdFromSceneId(string sceneId)
+        {
+            // This case happens when loading worlds
+            if (sceneId.StartsWith(URN_PREFIX))
+            {
+                int prefixLength = URN_PREFIX.Length;
+                return sceneId.Substring(prefixLength, sceneId.IndexOf("?", StringComparison.Ordinal) - prefixLength);
+            }
+
+            return sceneId;
         }
 
```

Comment:
```suggestion
                return dropBaseUrlFromSceneId(sceneId);
            }

            return sceneId;
        }

        /**
         * Drops the "baseUrl" argument out of Worlds-based content elements. `urn:decentraland:entity:bafkreicdminb5uh74p3p3htt5qnzcg5r5lti3fx5pwlv56h4wyckvtndc4?baseUrl=https://worlds-content-server.decentraland.org/contents/` becomes just `urn:decentraland:entity:bafkreicdminb5uh74p3p3htt5qnzcg5r5lti3fx5pwlv56h4wyckvtndc4`
         */
        private string dropBaseUrl(string sceneId) {
            int prefixLength = URN_PREFIX.Length;
            int questionMarkPosition = sceneId.IndexOf("?", StringComparison.Ordinal);
            return sceneId.Substring(prefixLength, questionMarkPosition - prefixLength);
        }
```

Reply:
I think the method name `GetEntityIdFromSceneId` is pretty much self-explanatory, I added that small additional comment to empathize that that condition is worlds-related